flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/PartitionableSourceITCase.scala

```diff
@@ -120,6 +127,33 @@ class PartitionableSourceITCase(
       )
     )
   }
+
+  @Test
+  def testPartitionPrunerCompileClassLoader(): Unit = {
+    val udfJavaCode =
+      s"""
+        |public class TrimUDF extends org.apache.flink.table.functions.ScalarFunction {
+        |   public String eval(String str) {
+        |     return str.trim();
+        |   }
+        |}
+        |""".stripMargin
+    val tmpDir: File = TEMPORARY_FOLDER.newFolder()
+    val udfJarFile: File = TestUserClassLoaderJar.createJarFile(
+      tmpDir, "flink-test-udf.jar", "TrimUDF", udfJavaCode)
+    val jars: util.List[URL] = new util.ArrayList[URL]()
+    jars.add(udfJarFile.toURI.toURL)
+    val cl = ClientUtils.buildUserCodeClassLoader(jars, java.util.Collections.emptyList(),
```

Comment:
nit: For lists with single constant element use `java.util.Collections.singletonList`.